### ISSUE TITLE
User invitations?

### ISSUE THREAD

**Issue by bnaul** [skyportal/skyportal#54]
Currently if someone has the URL for a skyportal instance, they could log in via Google but wouldn't be able to see any groups/sources/anything. Is this the behavior we want, or should access be limited to people who have been "invited" somehow?

**Comment by stefanv** [comment]
@profjsb Can you make a call on this issue?

**Comment by profjsb** [comment]
**TL;DR**: I think it's fine to leave the flow like it is for the time being (ie for milestone 0.1). But we should consider the login workflow more carefully in later milestones.

The main issue as I see it is the low barrier to entry for someone to be able to log in and consume (albeit small) resources, even if they cannot see any streams. (Indeed we may want to have some public streams globally viewable, so that all authed users can see such data). If a bot got loose in our system it would be painful to fix.

It seems like normal practice in SaaS services is to have a new user at least "get an account" with us, to say something about themselves before they can auth in. Even if we use Google auth we still want to know more about this person (Name, Institution, Role/Job function). Probably add Captcha just to make it semi-safer.

We could set a default public group (e.g. Public ZTF) so that a new User can see real data right away. How a User is then added to another set of groups is more of an open question. Some possibilities:

   1) On the User "My Account" page, there is a checklist of groups to join, then admin of each group gets an email with a unique link to ok the adding of that person. Negative: this opens up the private group names.

   2) Ask User to email group admins to request permissions. Makes this process more secure but is less slick, and adds an extra burden to do anything substantive.

To be sure, for an uninitiated DB (ie. first use), we should set up the initial (google-based) email to be superadmin.

**Comment by dannygoldstein** [comment]
Fixed by #757
